### PR TITLE
12.0 mig barcodes generator abstract

### PR DESCRIPTION
**Pull request by cvinh**
works out of the box in v12

### CODE REVIEW

**Review by phatnguyenuit — CHANGES_REQUESTED** [review]
Missing file `requirements.txt` at the top level of this repository. Refer the old one to get it.

File: barcodes_generator_abstract/models/barcode_generate_mixin.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+try:
+    import barcode
+except ImportError:
+    _logger.debug("Cannot import 'viivakoodi' python library.")
+    barcode = None
+
+
+class BarcodeGenerateMixin(models.AbstractModel):
+
+    _name = 'barcode.generate.mixin'
```
Comment (pedrobaeza) [inline barcodes_generator_abstract/models/barcode_generate_mixin.py:23]: Add `_description` for avoiding runbot warning

**Review by pedrobaeza — COMMENTED** [review]
And you should check #193 for an alternative library.

File: barcodes_generator_abstract/models/barcode_generate_mixin.py (hunk 1 above)
Comment (cvinh) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py:23]: About the library, tell me if I should change it
About _description, I will do a force push tomorrow

Comment (pedrobaeza) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py:23]: @legalsylvain @yucer what do you think?

Comment (legalsylvain) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py:23]: Hi. As said in #193, I have no point of view, regarding PIL / Pillow neither regarding the quality of the code of both librairies.
But the othe lib has more contributors. So I'm in favour to switch, if it works correctly. 

@cvinh, could you test and check if all is OK ? 

regards.

Comment (cvinh) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py:23]: @pedrobaeza could you help with the runbot error ? I don't see where it fails... thanks in advance

Comment (cvinh) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py:23]: @gurneyalex could you help with the runbot error ? I don't see where it fails... thanks in advance

**Comment by pedrobaeza** [comment]
The warning is:
```
2019-08-13 23:03:45,710 161 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (sequence_id, sequence) of barcode.rule() have the same label: Sequence.
```

**Review by phatnguyenuit — APPROVED** [review]

**Review by legalsylvain — APPROVED** [review]
/ocabot merge

LGTM. Thanks for porting this module.

File: barcodes_generator_abstract/models/barcode_generate_mixin.py (hunk 1 above)
Comment (yucer) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py:23]: @pedrobaeza sorry for the late response. I am not too much familiarized with these modules. Just noted that there were two alternate python libraries for the same feature, and the one not used had more activity.

Thanks for the feed back.

Comment (pedrobaeza) [reply barcodes_generator_abstract/models/barcode_generate_mixin.py:23]: Module was already merge, so a bit late I'm afraid. Any way, we have switched the library.

**Pull request merged**